src/entries/popup/components/SwitchMenu/SwitchNetworkMenu.tsx

```diff
@@ -57,9 +59,18 @@ export const SwitchNetworkMenuSelector = ({
   disconnect?: () => void;
   onNetworkSelect?: (event?: Event) => void;
   onShortcutPress: (chainId: string) => void;
+  onlySwapSupportedNetworks?: boolean;
 }) => {
   const { trackShortcut } = useKeyboardAnalytics();
-  const { chains } = useUserChains();
+  const { chains: userChains } = useUserChains();
+  const { customChains } = useCustomNetwork();
+
+  const chains = useMemo(() => {
+    const customChainsIds = customChains.map((chain) => chain.id);
+    return onlySwapSupportedNetworks
+      ? userChains.filter((chain) => !customChainsIds.includes(chain.id))
+      : userChains;
```

Comment:
don't we have an array of `supportedChains` somewhere? like when onlySwapSupportedNetworks true returns the SUPPORTED_CHAINS array constant, hmm have to filter the ones hidden idk

Reply:
yeah we do have SUPPORTED_CHAINS but it includes all testnets